veniq/dataset_collection/dataflow/main.py

```diff
@@ -0,0 +1,123 @@
+import os
+from functools import lru_cache
+from functools import partial
+from pathlib import Path
+from typing import Dict, Tuple
+
+import bonobo
+
+from veniq.dataset_collection.dataflow.data_aggregation import aggregate
+from veniq.dataset_collection.dataflow.filter_invalid_cases import filter_invalid_cases
+from veniq.dataset_collection.dataflow.save_file import save_input_file, save_output_file
+from veniq.dataset_collection.dataflow.preprocess import preprocess, create_existing_dir
+from veniq.dataset_collection.dataflow.annotation import annotate
+from veniq.dataset_collection.dataflow.find_inline_methods import find_EMs
+from veniq.dataset_collection.dataflow.inlining import inline
+from veniq.dataset_collection.dataflow.ncss_filter import filter_by_ncss
+
+
+def get_list_of_files(path: str):
+    test_files = set(Path(path).glob('**/*Test*.java'))
+    not_test_files = set(Path(path).glob('**/*.java'))
+    for x in not_test_files.difference(test_files):
+        yield x
+
+
+@lru_cache()
+def create_dirs(output):
+    full_dataset_folder = Path(output) / 'full_dataset'
+    output_dir = full_dataset_folder / 'output_files'
+    create_existing_dir(output_dir)
+    input_dir = full_dataset_folder / 'input_files'
+    create_existing_dir(input_dir)
+    return {'input_dir': str(input_dir), 'output_dir': str(output_dir)}
+
+
+def pass_params_to_next_node(dirs_dict, em_item: Dict[Tuple, Tuple]):
```

Comment:
come up with a custom type for the contents of the em_item dictionary? it's hard to interpret otw